### PR TITLE
TRUNK-6070 Back port to 2.5.x [Add Patient Program to Patient Identifier]

### PR DESCRIPTION
**Pull request by slubwama**
Back Port to 2.5.x
Ticket
https://issues.openmrs.org/browse/TRUNK-6070
Add a patient program to the patient identifier to create a connection between a patient identifier and a patient program
Create a method to allow getting a patient identifier by a patient program.

PR to Master https://github.com/openmrs/openmrs-core/pull/4044

### CODE REVIEW

**Comment by kdaud** [comment]
These are just some minor fixes

File: api/src/main/java/org/openmrs/PatientIdentifier.java [hunk 1]
```diff
@@ -66,6 +66,9 @@ public class PatientIdentifier extends BaseChangeableOpenmrsData implements java
 	
 	private Location location;
 
+	private PatientProgram patientProgram;
+	
+
```
Comment (kdaud) [inline api/src/main/java/org/openmrs/PatientIdentifier.java:71]:
```suggestion
```

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java [hunk 2]
```diff
@@ -993,4 +994,15 @@ public Allergy saveAllergy(Allergy allergy) {
     	sessionFactory.getCurrentSession().save(allergy);
     	return allergy;
     }
+
```
Comment (kdaud) [inline api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java:997]:
```suggestion
```

Comment (slubwama) [reply api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java:997]: @kdaud done

File: api/src/main/java/org/openmrs/PatientIdentifier.java [hunk 3]
```diff
@@ -328,4 +331,23 @@ public int compare(PatientIdentifier pi1, PatientIdentifier pi2) {
 			return retValue;
 		}
 	}
+
```
[the remark this answers is not in the document]
Comment (slubwama) [reply api/src/main/java/org/openmrs/PatientIdentifier.java:333]: @kdaud done

**Review by kdaud — APPROVED** [review]
Thanks @slubwama 

It would be a good deal to have `pull/4070` updated as well.

**Comment by slubwama** [comment]
> Thanks @slubwama
> 
> It would be a good deal to have `pull/4070` updated as well.

Done

**Comment by kdaud** [comment]
I meant the changes suggested here are also suitable for the pr with ID `pull/4070` that Backport 2.4.x branch.

**Comment by slubwama** [comment]
> I meant the changes suggested here are also suitable for the pr with ID `pull/4070` that Backports 2.4.x branch.

I understood that and I completed that

**Pull request closed without merging**